enterprise/signals.py

```diff
@@ -341,6 +392,37 @@ def create_enterprise_enrollment_receiver(sender, instance, **kwargs):     # pyl
         )
 
 
-# Don't connect this receiver if we dont have access to CourseEnrollment model
+def generated_certificate_receiver(sender, instance, **kwargs):  # pylint: disable=unused-argument
+    """
+    Watches for post_save signal for creates/updates on the GeneratedCertificate table.
+
+    Filters EnterpriseCourseEnrollments to find any enrollments associated with the given user and
+    course. If found, an activity stream action is sent to indicate the user earned a certificate. 🥳
+    This action is only sent if the EnterpriseCustomerUser is an active community member and the
+    community feature is enabled for the associated EnterpriseCustomer.
+    """
+    if CertificateStatuses.is_passing_status(instance.status):
+        user = instance.user
+        enterprise_enrollments = EnterpriseCourseEnrollment.objects.filter(
+            enterprise_customer_user__user_id=user.id,
+            course_id=instance.course_id,
+        )
+        for enterprise_enrollment in enterprise_enrollments:
```

Comment:
this makes it seem like there could be multiple enrollments made at once - is that ever a possibility? It might make more sense to order the results of the query by `created` (descending) and pick out the first object